galera/src/certification.cpp

```diff
@@ -118,6 +118,17 @@ galera::Certification::purge_for_trx_v3(TrxHandle* trx)
     for (long i = 0; i < keys.count(); ++i)
     {
         const KeySet::KeyPart& kp(keys.next());
+
```

Comment:
I am wondering what causes local key to empty. My investigation few days back suggested that only if you append_key trx is forwarded for certification.

Purge action explain the intermediate failure.

Can you write a small tcs where-in you see local trx generating an empty key part.